test/unit/math/rev/mat/functor/algebra_solver_test.cpp

```diff
@@ -312,3 +339,205 @@ TEST(MathMatrix, system_functor_constructor) {
+}
+
+TEST_F(error_message_test, newton) {
+  using stan::math::var;
+  bool is_newton = true;
+  Eigen::Matrix<var, Eigen::Dynamic, 1> y = y_2;
+  error_conditions_test(non_linear_eq_functor(), y, is_newton);
+}
+
+TEST_F(error_message_test, newton_dbl) {
+  bool is_newton = true;
+  error_conditions_test(non_linear_eq_functor(), y_3, is_newton);
+}
+
+TEST_F(max_steps_test, newton) {
+  bool is_newton = true;
+  max_num_steps_test(y_var, is_newton);
+}
+
+TEST_F(max_steps_test, newton_dbl) {
+  bool is_newton = true;
+  max_num_steps_test(y, is_newton);
+}
+
+TEST(MathMatrix, unsolvable_flag_newton) {
+  Eigen::Matrix<stan::math::var, Eigen::Dynamic, 1> y(2);
+  y << 1, 1;
+
+  unsolvable_flag_test(y);
+}
+
+TEST(MathMatrix, unsolvable_flag_newton_dbl) {
+  Eigen::VectorXd y(2);
+  y << 1, 1;
+
+  unsolvable_flag_test(y);
+}
+
+TEST_F(degenerate_eq_test, newton) {
+  using stan::math::algebra_solver_powell;
+  using stan::math::sum;
+  using stan::math::var;
+
+  // This first initial guess produces the
+  // solution x = {8, 8}
+  bool is_newton = true;
+  for (int k = 0; k < n_x; k++) {
+    Eigen::Matrix<var, Eigen::Dynamic, 1> y = y_dbl;
+    Eigen::Matrix<var, Eigen::Dynamic, 1> theta
+        = degenerate_test(y, x_guess_1, is_newton);
+    EXPECT_FLOAT_EQ(8, theta(0).val());
+    EXPECT_FLOAT_EQ(8, theta(1).val());
+
+    AVEC y_vec = createAVEC(y(0), y(1));
+    VEC g;
+    theta(k).grad(y_vec, g);
+
+    for (int l = 0; l < n_y; l++)
+      EXPECT_NEAR(J1(k, l), g[l], tolerance);
+  }
+
+  // This next initial guess produces the
+  // solution x = {5, 5}
+  for (int k = 0; k < 1; k++) {
+    Eigen::Matrix<var, Eigen::Dynamic, 1> y = y_dbl;
+    Eigen::Matrix<var, Eigen::Dynamic, 1> theta
+        = degenerate_test(y, x_guess_2, is_newton);
+    EXPECT_FLOAT_EQ(5, theta(0).val());
+    EXPECT_FLOAT_EQ(5, theta(0).val());
+
+    AVEC y_vec = createAVEC(y(0), y(1));
+    VEC g;
+    theta(k).grad(y_vec, g);
+
+    for (int l = 0; l < n_y; l++)
+      EXPECT_NEAR(J2(k, l), g[l], tolerance);
+  }
+}
+
+TEST_F(degenerate_eq_test, newton_dbl) {
+  bool is_newton = true;
+
+  // This first initial guess produces the
+  // solution x = {8, 8}
+  Eigen::VectorXd theta = degenerate_test(y_dbl, x_guess_1, is_newton);
+  EXPECT_FLOAT_EQ(8, theta(0));
+  EXPECT_FLOAT_EQ(8, theta(1));
+
+  // This next initial guess produces the
+  // solution x = {5, 5}
+  theta = degenerate_test(y_dbl, x_guess_2, is_newton);
+  EXPECT_FLOAT_EQ(5, theta(0));
+  EXPECT_FLOAT_EQ(5, theta(1));
+
+  // See if the initial guess determines neighborhood of the
+  // solution, when solutions have different scales,
+  // using y_scale.
+  // theta = degenerate_test(y_scale, x_guess_2, is_newton);
+  // EXPECT_FLOAT_EQ(5, theta(0));
+  // EXPECT_FLOAT_EQ(5, theta(1));
+  //
+  // theta = degenerate_test(y_scale, x_guess_3, is_newton);
+  // EXPECT_FLOAT_EQ(100, theta(0));
+  // EXPECT_FLOAT_EQ(100, theta(1));
+  //
+  // NOTE: causes Newton solver to send a flag -11 error.
```

Comment:
Could you investigate this? It doesn't look like something we should ignore.

Reply:
My guess is it's related to Jacobian calculation with `x_guess_3`. Maybe try removing `KINSetJacFn` to use default DQ method and see if the error persists, and start from there.

Reply:
The error persists when using DQ. I also tried remove the linesearch step, and forcing the algorithm to evaluate the Jacobian at each step.

Reply:
See comment here: https://discourse.mc-stan.org/t/newton-solver-in-stan/6684/68.

If the initial guess is (80, 80) the solver converges to one of the roots.

Reply:
Solved -- see discussion (https://discourse.mc-stan.org/t/newton-solver-in-stan/6684/86).